off_nitrogens/perturb_angle.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+
+#=============================================================================================
+# MODULE DOCSTRING
+#=============================================================================================
+
+"""
+perturb_angle.py
+
+Perturb geometry around an improper dihedral angle. The options are:
+ 1. Change valence angle without changing improper
+ 2. Change improper angle without changing valence angles
+
+For use in generating geometries to parameterize valence and improper angles
+
+By: Victoria Lim and Jessica Maat
+
+"""
+
+#=============================================================================================
+# GLOBAL IMPORTS
+#=============================================================================================
+
+import numpy as np
+import math
+import sys
+
+from calc_improper import *
+
+#=============================================================================================
+# PRIVATE SUBROUTINES
+#=============================================================================================
+
+def rotation_matrix(axis, theta):
+    """
+    Return the rotation matrix associated with counterclockwise rotation about
+    the given axis by theta radians.
+
+    Parameters
+    ----------
+    axis : tuple, list, or numpy array
+        vector normal to plane in which rotation is performed
+    theta : float
+        angle of rotation amount in degrees
+
+    Returns
+    -------
+    numpy array
+        three-dimension rotation matrix
+
+    Reference
+    ---------
+    https://tinyurl.com/yam5hu78
+    """
+    axis = np.asarray(axis)
+    theta = np.radians(theta)
+
+    axis = axis/math.sqrt(np.dot(axis, axis))
+    a = math.cos(theta/2.0)
+    b, c, d = -axis*math.sin(theta/2.0)
+    aa, bb, cc, dd = a*a, b*b, c*c, d*d
+    bc, ad, ac, ab, bd, cd = b*c, a*d, a*c, a*b, b*d, c*d
+    return np.array([[aa+bb-cc-dd, 2*(bc+ad), 2*(bd-ac)],
+                     [2*(bc-ad), aa+cc-bb-dd, 2*(cd+ab)],
+                     [2*(bd+ac), 2*(cd-ab), aa+dd-bb-cc]])
```

Comment:
This is one of those things where I'm always sure numpy or scipy would already have the function, but I guess not haha